fhir/com.b2international.snowowl.fhir.rest/src/com/b2international/snowowl/fhir/rest/MetadataController.java

```diff
@@ -0,0 +1,387 @@
+									.url("apiVersion")
+									.value("Snow Owl API Version comes here")
+									.build())
+								.build())
+						.build())
+				.implementation(Implementation.builder()
+						.url("b2i.sg")
+						.description("Demo server")
+						.build())
+				.addFormat(new Code(AbstractFhirResourceController.APPLICATION_FHIR_JSON))
+				.addRest(restBuilder.build())
+				.build();
+		
+	}
+	
+	private Collection<Resource> collectResources(OpenAPI openAPI) {
+		
+		Paths paths = openAPI.getPaths();
+		
+		List<io.swagger.v3.oas.models.tags.Tag> tags = openAPI.getTags();
+		
+		//Class-level tags with extensions indicate a resource
+		Set<io.swagger.v3.oas.models.tags.Tag> fhirTags = tags.stream()
+				.filter(t -> t.getExtensions() != null && t.getExtensions().containsKey(B2I_OPENAPI_X_NAME))
+				.collect(Collectors.toSet());
+		
+		@SuppressWarnings("unchecked")
+		List<Resource> resources = fhirTags.stream().map(t -> {
+			Map<String, String> nameExtensionMap = (Map<String, String>) t.getExtensions().get(B2I_OPENAPI_X_NAME);
+			
+			Resource.Builder resourceBuilder = Resource.builder()
+				.type(t.getName())
+				.profile((String)nameExtensionMap.get(B2I_OPENAPI_PROFILE));
+			
+			//Collect the operations that belong to the same tagged class resource
+			Set<io.swagger.v3.oas.models.Operation> commonOperations = paths.values().stream()
+					.flatMap(pi -> pi.readOperations().stream())
+					.filter(o -> o.getTags().contains(t.getName()))
+					.collect(Collectors.toSet());
+				
+			for (io.swagger.v3.oas.models.Operation op : commonOperations) {
+				Map<String, Object> operationExtensionMap = op.getExtensions();
+				
+				//skip the operations with no B2i extensions
+				if (operationExtensionMap == null || !operationExtensionMap.containsKey(B2I_OPENAPI_X_INTERACTION)) continue;
+				
+				Map<String, String> interactionMap = (Map<String, String>) operationExtensionMap.get(B2I_OPENAPI_X_INTERACTION);
+				
+				interactionMap.keySet().forEach(k -> {
+				
+					Interaction.Builder interactionBuilder = Interaction.builder()
+						.code((String) k);
+					
+					Object value = interactionMap.get(k);
+						
+					if (value != null && !((String) value).isEmpty()) {
+						interactionBuilder.documentation((String) value);
+					}
+					resourceBuilder.addInteraction(interactionBuilder.build());
+				});
+			}
+			return resourceBuilder.build();
+		}).collect(Collectors.toList());
+		
+		resources.sort((r1, r2) -> r1.getType().getCodeValue().compareTo(r2.getType().getCodeValue()));
+		return resources;
+	}
+
+	private String buildOperationUrl(HttpServletRequest request, Code code, OperationDefinition operationDefinition) {
+		
+		try {
+			URL url = new URL(request.getRequestURL().toString());
+			StringBuilder sb = new StringBuilder(url.getProtocol())
+					.append("://")
+					.append(url.getHost())
+					.append(":")
+					.append(url.getPort())
+					.append(java.nio.file.Paths.get(url.getPath()).getParent().toString())
```

Comment:
Using `java.nio.file.Paths` is incorrect here and might lead to incorrect path segments because the file path separator is different on certain platforms, like Windows vs Linux, Mac etc. Please refactor.